shimmingtoolbox/b1/b1_shim.py

```diff
@@ -0,0 +1,270 @@
+#!usr/bin/env python3
+# -*- coding: utf-8
+
+import numpy as np
+import scipy.optimize
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def b1_shim(b1_maps, mask, cp_weights=None, algo=1, reg_param=0, target=None,  q_matrix=None, sar_factor=1, constrained=False):
+    """
+    Computes static optimized shim weights that minimize the B1 field coefficient of variation over the masked region.
+
+    Args:
+        b1_maps (numpy.ndarray): 4D array  corresponding to the measured B1 field. (x, y, n_slices, n_channels)
+        mask (numpy.ndarray): 3D array corresponding to the region where shimming will be performed. (x, y, n_slices)
+        cp_weights (numpy.ndarray): 1D vector of length n_channels of complex weights corresponding to the CP mode of
+        the coil. Must be normalized.
+        algo (int): Number from 1 to 5 specifying which algorithm to use for B1 optimization:
+                    1 - Optimization aiming to reduce the coefficient of variation (CoV) of the resulting B1+ field.
```

Comment:
maybe consider merging the optim functions for usage with B0 shimming. Also tagging @po09i